medusa/tv/episode.py

```diff
@@ -1970,3 +2018,66 @@ def airdate_modify_stamp(self):
                     'location': os.path.basename(self.location),
                 }
             )
+
+    def update_status(self, filepath):
+        """Update the episode status according to the file information.
+
+        The status should only be changed if either the size or the filename changed.
+        :param filepath: Path to the new episode file.
+        """
+        old_status, old_quality = Quality.split_composite_status(self.status)
+
+        old_location = self.location
+        # Changing the name of the file might also change its quality
+        same_name = old_location and os.path.normpath(old_location) == os.path.normpath(filepath)
+
+        old_size = self.file_size
+        # Setting a location to episode, will get the size of the filepath
+        with self.lock:
+            self.location = filepath
+        # If size from given filepath is 0 it means we couldn't determine file size
+        same_size = old_size and self.file_size > 0 and self.file_size == old_size
+
+        if not same_size or not same_name:
+            log.debug(
+                '{name}: The old episode had a different file associated with it, '
+                're-checking the quality using the new filename {filepath}',
+                {'name': self.series.name, 'filepath': filepath}
+            )
+
+            new_quality = Quality.name_quality(filepath, self.series.is_anime)
+
+            if old_status in (SNATCHED, SNATCHED_PROPER, SNATCHED_BEST) or (
+                    old_status == DOWNLOADED and old_location):
+                new_status = DOWNLOADED
+            else:
+                new_status = ARCHIVED
```

Comment:
This means everything else will be tagged as `Archived` when refreshing a show,
meaning episodes with `Wanted` or `Failed` status and a file path would get tagged as `Archived`.
That is something to consider.
@medariox and I tried to wrap our heads around it, but ultimately I think it's just a decision that needs to be made...

Reply:
What should be added is: this situation can only happen if the file was `manually` changed. Just triggering a refresh of the show won't change its status. Also now the status goes from WANTED to ARCHIVED only if an episode was already previously present. Related tests: https://github.com/pymedusa/Medusa/pull/4157/files#diff-fe5305567785e70fe5a0a06240cbcb5fR24